exercises/collatz-conjecture/canonical-data.json

```diff
@@ -0,0 +1,48 @@
+{
+  "exercise": "collatz-conjecture",
+  "version": "1.0.0",
+  "cases": [
+    {
+      "description": "1 is 0 steps",
+      "property": "collatz",
+      "number": 1,
+      "expected": "0"
+    },
+    {
+      "description": "2 is 1 step",
+      "property": "collatz",
+      "number": 2,
+      "expected": "1"
+    },
+    {
+      "description": "9 is 19 steps",
+      "property": "collatz",
+      "number": 9,
+      "expected": "19"
+    },
+    {
+      "description": "27 is 111 steps",
+      "property": "collatz",
+      "number": 27,
+      "expected": "111"
+    },
+    {
+      "description": "97 is 118 steps",
+      "property": "collatz",
+      "number": 97,
+      "expected": "118"
+    },
+    {
+      "description": "871 is 178 steps",
+      "property": "collatz",
+      "number": 871,
+      "expected": "178"
+    },
+    {
+      "description": "6171 is 261 steps",
```

Comment:
I initially pulled all the values from the Wikipedia, but I had the same thought and changed the test cases to cover the n/2 and 3n+1 choices somewhat separately. I also added tests to enforce only positive numbers are valid input.